### ISSUE TITLE
zstd support?

### ISSUE THREAD

**Issue by hmaarrfk** [h5netcdf/h5netcdf#240]
Does h5netcdf support zstd compression?

#### Is your feature request related to a problem? Please describe.
```
import xarray as xr
ds = xr.DataArray(np.zeros(3), name='data).to_dataset()
ds['data'].encoding['compression'] = 'zstd'
# netcdf4 works
ds.to_netcdf('test.nc', engine='netcdf4')

# h5netcdf doesn't
ds.to_netcdf('test.nc', engine='h5netcdf')
```
```
ValueError: Compression filter "zstd" is unavailable
```

#### Describe the solution you'd like
Just to know if there is a path forward at all...

#### Describe alternatives you've considered
netcdf4. But we've found that it has some strange bugs in our workflows....

#### Additional context
I can help compile things on conda-forge.

**Comment by kmuehlbauer** [comment]
@hmaarrfk `h5netcdf` should already be able to use [`hdf5plugin`](http://www.silx.org/doc/hdf5plugin/latest/) out of the box. How about:

```python
import numpy as np
import xarray as xr
import hdf5plugin
ds = xr.DataArray(np.zeros(3), name='data').to_dataset()
ds['data'].encoding['compression'] = hdf5plugin.Zstd()
# h5netcdf works
ds.to_netcdf('test.nc', engine='h5netcdf')
# netcdf4 doesn't
ds.to_netcdf('test.nc', engine='netcdf4')
```

Seems to read just fine for both, so I'm assuming there is some incantation to make `hdf5plugin` usable for `netcdf4-python` on write the same way.

```python
with xr.open_dataset("test.nc", engine="h5netcdf") as h5ds:
    with xr.open_dataset("test.nc", engine="netcdf4") as ncds:
        xr.testing.assert_equal(h5ds, ncds)
```

**Comment by hmaarrfk** [comment]
Hmm that does seem to look like it works. But it is definitely "clunky" in the sense that I must know what my backend is going to be do to it right.

I guess some PRs could be made to xarray to help tie in the string `"zstd"` to `hdf5plugin.Zstd()` when the h5netcdf plugin is used.

Thanks1

**Comment by kmuehlbauer** [comment]
@hmaarrfk I agree, it would be great to have a unified API. We could also implement the netcdf4-python behavior into h5netcdf until there is a more generic approach.

**Comment by hmaarrfk** [comment]
I think it might be a combination of both.

I would have to study how xarray communicates with netcdf4 more.

This is the PR i was thinking about, but hte xarray portion seems quite thin, it seems maybe a feature of netcdf4.
https://github.com/pydata/xarray/pull/7551/files#diff-d8ee99661100079ac05015798bf29a001ed88af1716a33a1ec3c727cf554b949

**Comment by kmuehlbauer** [comment]
OMG, it's even easier than expected with `hdf5plugin`:

```python
In [4]: dict(**hdf5plugin.Blosc(cname="lz4hc", clevel=5, shuffle=1))
Out[4]: {'compression': 32001, 'compression_opts': (0, 0, 0, 0, 5, 1, 2)}
```
